### PR TITLE
Fixing - IDENTITY-3834

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
https://wso2.org/jira/browse/IDENTITY-3834

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 1]
```diff
@@ -104,90 +107,103 @@ public static IdentityProvider buildeFederatedIdentityProvider(HttpServletReques
...
+                        idpUUID = diskFileItem.getString();
+                    }
+
+                    if ("certFile".equals(key)) {
+                        paramMap.put(key, Base64.encode(value));
+                    } else if ("google_prov_private_key".equals(key)) {
+                        paramMap.put(key, Base64.encode(value));
+                    } else if (key.startsWith("claimrowname_")) {
+                        idpClaims.add(new String(value));
+                        paramMap.put(key, new String(value));
```
Comment (pulasthi7) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: better to assign to a variable and then add/put without creating two string objects

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 2]
```diff
@@ -104,90 +107,103 @@ public static IdentityProvider buildeFederatedIdentityProvider(HttpServletReques
...
+                    } else if (key.startsWith("rolerowname_")) {
+                        idpRoles.add(new String(value));
+                        paramMap.put(key, new String(value));
```
Comment (pulasthi7) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: better to assign to a variable and then add/put without creating two string objects

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 3]
```diff
@@ -195,12 +196,21 @@
...
 
     String[] idpClaims = new String[]{"admin", "Internal/everyone"};//appBean.getSystemClaims();
 
-    IdentityProvider identityProvider = (IdentityProvider) session.getAttribute("identityProvider");
-    List<IdentityProvider> identityProvidersList = (List<IdentityProvider>) session.getAttribute("identityProviderList");
+    Map<String, UUID> idpUniqueIdMap = (Map<String, UUID>) session.getAttribute("idpUniqueIdMap");
+
+    IdentityProvider identityProvider = null;
+
+    if(idPName != null && idpUniqueIdMap.get(idPName) != null) {
```
Comment (pulasthi7) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: better to have a null check on 'idpUniqueIdMap'.

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp [hunk 4]
```diff
@@ -41,7 +43,8 @@
             IdentityProviderMgtServiceClient client = new IdentityProviderMgtServiceClient(cookie, backendServerURL, configContext);
             IdentityProvider identityProvider = client.getIdPByName(idPName);
             if(identityProvider != null){
-                session.setAttribute("identityProvider", identityProvider);
+                Map<String, UUID> idpUniqueIdMap = (Map<String, UUID>) session.getAttribute("idpUniqueIdMap");
+                session.setAttribute(idpUniqueIdMap.get(idPName).toString(), identityProvider);
```
Comment (pulasthi7) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp]: check 'idpUniqueIdMap.get(idPName)' for null.

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java (hunk 1 above)
Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java (hunk 2 above)
Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp (hunk 3 above)
Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp (hunk 4 above)
Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 5]
```diff
@@ -104,90 +107,103 @@ public static IdentityProvider buildeFederatedIdentityProvider(HttpServletReques
...
+            for (Object item : items) {
+
+                DiskFileItem diskFileItem = (DiskFileItem) item;
+
+                if (diskFileItem != null) {
+
+                    byte[] value = diskFileItem.get();
+                    String key = diskFileItem.getFieldName();
+
+                    if(StringUtils.equals(key, "idpUUID")) {
```
Comment (chrishantha) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Format code

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 6]
```diff
@@ -104,90 +107,103 @@ public static IdentityProvider buildeFederatedIdentityProvider(HttpServletReques
...
+                    if (updatedValue != null && updatedValue.trim().length() == 0) {
+                        paramMap.put(key, null);
                     }
                 }
             }
 
+            IdentityProvider oldIdentityProvider = (IdentityProvider) request.getSession().getAttribute(idpUUID);
+
+            if(oldIdentityProvider != null) {
+                if(oldIdpName == null) {
```
Comment (chrishantha) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Format code

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp [hunk 7]
```diff
@@ -41,7 +43,8 @@
...
             }
         } else if(session.getAttribute("identityProviderList") == null){
```
Comment (chrishantha) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp]: Format code

Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit-load.jsp]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java (hunk 6 above)
Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java (hunk 5 above)
Comment (JKAUSHALYA) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

**Pull request merged**